`TestAzureWatcher/filter_by_resource_group` flakiness

## Failure

#### Link(s) to logs

https://console.cloud.google.com/cloud-build/builds/a57ad039-b1a9-41b8-86b9-78803ac498b9;step=0?project=ci-account

#### Relevant snippet

```
=== RUN   TestAzureWatcher/filter_by_resource_group
    azure_watcher_test.go:150: 
        	Error Trace:	/workspace/lib/srv/server/azure_watcher_test.go:150
        	Error:      	elements differ
        	            	
        	            	extra elements in list B:
        	            	([]interface {}) (len=1) {
        	            	 (string) (len=3) "vm5"
        	            	}
        	            	
        	            	
        	            	listA:
        	            	([]string) (len=2) {
        	            	 (string) (len=3) "vm1",
        	            	 (string) (len=3) "vm2"
        	            	}
        	            	
        	            	
        	            	listB:
        	            	([]string) (len=3) {
        	            	 (string) (len=3) "vm5",
        	            	 (string) (len=3) "vm1",
        	            	 (string) (len=3) "vm2"
        	            	}
        	Test:       	TestAzureWatcher/filter_by_resource_group
    --- FAIL: TestAzureWatcher/filter_by_resource_group (0.00s)
===================================================
```

--- a/lib/services/matchers.go
+++ b/lib/services/matchers.go
@@ -70,7 +70,7 @@ type AzureMatcher struct {
 	ResourceGroups []string
 	// Types are Azure resource types to match, for example "mysql" or "postgres".
 	Types []string
-	// Regions are Azure regions to query for databases.
+	// Regions are Azure regions to query for resources.
 	Regions []string
 	// ResourceTags are Azure tags to match.
 	ResourceTags types.Labels

--- a/lib/srv/server/azure_watcher_test.go
+++ b/lib/srv/server/azure_watcher_test.go
@@ -99,7 +99,7 @@ func TestAzureWatcher(t *testing.T) {
 				Regions:        []string{"location1", "location2"},
 				ResourceTags:   types.Labels{"*": []string{"*"}},
 			},
-			wantVMs: []string{"vm1", "vm2"},
+			wantVMs: []string{"vm1", "vm2", "vm5"},
 		},
 		{
 			name: "filter by location",

